homeassistant/components/econet/water_heater.py

```diff
@@ -40,6 +40,11 @@
 ATTR_START_DATE = "start_date"
 ATTR_END_DATE = "end_date"
 
+ATTR_LOWER_TEMP = "lower_temp"
+ATTR_UPPER_TEMP = "upper_temp"
+ATTR_AMBIENT_TEMP = "ambient_temp"
```

Comment:
Ambient temp sounds like it should be a separate sensor platform. Then we need to first refactor this integration to move the setup and config to the integration package module and set up platforms via discovery helper `load_platform`.

Reply:
It's the ambient temperature around the hot water heater and the sensor is inside the hot water heater (it's reading the temperature of the air being sucked into the heat pump). Not sure why that would be a separate sensor platform?

Reply:
The reason is that the value is relevant also without setting up the water heater platform. The ambient temperature reading of the room is relevant on its own. The other three attributes are not relevant if we don't set up the water heater platform.

Reply:
ok - I don't have the time or resources to make huge implementations like that right now and I'd really like to at least have the other three attributes right now. So I guess the best option is for me to close this pull request and submit a new one with just the other three in it?

Reply:
It would be ok to just remove the ambient attribute from this PR.

Reply:
I've been following this because I am using ambient temp in math along with upper to pick a setting to reduce total cost... 

Anyway, ambient temp isn't really a good tool for "room temp" because in a lot (maybe most) of implementations the value of the air as the water heater reads it is based on the air being pulled in by the fan on top, which is actually ductwork going to a different location.

The use case outside of the water heater itself is completely dependent on installation style and at best indirectly useful for another purpose, which seems to suggest an attribute as it is in this PR would be fine?

Reply:
![image](https://user-images.githubusercontent.com/59674699/78997252-cbaa1700-7b0b-11ea-8a32-32e07a9c0ee2.png)

Here is the "typical install" from the install guide. Note the ductwork on top labelled "vent".

Reply:
![image](https://user-images.githubusercontent.com/59674699/78997685-a79b0580-7b0c-11ea-8dea-a763656e6678.png)

Install examples from the manual. Note the ductwork at top is for air in to the device (which is what the water heater considers ambient temp) and the side ductwork is to dump the cold air out of the device.

Reply:
It's still reading the temperature of air coming from outside the device so is relevant on its own.